src/asset-relocator.js

```diff
@@ -462,25 +463,55 @@ module.exports = async function (content) {
       // require
       else if (!isESM && isRequire(node)) {
         const expression = node.arguments[0];
-        // require('bindings')('asdf')
         if (isStaticRequire(node) &&
             parent.type === 'CallExpression' &&
             parent.callee === node &&
-            expression &&
-            expression.value === 'bindings') {
-          let staticValue = computeStaticValue(parent.arguments[0], true);
-          let bindingsValue;
-          if (staticValue && 'value' in staticValue) {
-            try {
-              bindingsValue = createBindings()(staticValue.value);
+            expression) {
+          // require('bindings')('asdf')
+          if (expression.value === 'bindings') {
+            let staticValue = computeStaticValue(parent.arguments[0], true);
+            let bindingsValue;
+            if (staticValue && 'value' in staticValue) {
+              try {
+                bindingsValue = createBindings()(staticValue.value);
+              }
+              catch (err) {}
+            }
+            if (bindingsValue) {
+              staticChildValue = { value: bindingsValue };
+              staticChildNode = parent;
+              staticChildValueBindingsInstance = staticBindingsInstance;
+              return this.skip();
             }
-            catch (err) {}
           }
-          if (bindingsValue) {
-            staticChildValue = { value: bindingsValue };
-            staticChildNode = parent;
-            staticChildValueBindingsInstance = staticBindingsInstance;
-            return this.skip();
+          // require('pkginfo')(module, ...string[])
+          else if (expression.value === 'pkginfo' &&
+                  parent.arguments.length &&
+                  parent.arguments[0].type === 'Identifier' &&
+                  parent.arguments[0].name === 'module') {
+            let filterValues = [];
+            for (let i = 1; i < parent.arguments.length; i++) {
+              if (parent.arguments[i].type === 'Literal')
+                filterValues.push(parent.arguments[i].value);
+            }
+            const scope = getPackageScope(id);
+            if (scope) {
+              try {
+                var pkg = JSON.parse(readFileSync(scope + '/package.json'));
```

Comment:
Why `readFileSync`? Could you use promisify(readFile) and await it here?

Reply:
This is an AST walking routine, which is specifically a sync routine, so we can't make individual parts of the walker async.

Because this only applies just to this specific scenario, the cost doesn't seem restrictive, but the alternative could be a generic work queue.